docs/development.md

```diff
@@ -6,57 +6,105 @@ nav_order: 8
 
+Tasks can be listed by running `task --list-all`.
 
-## Before committing
+## Running tests
 
-Install [Taskfile](https://taskfile.dev/installation/) to run tasks locally. Tasks can be listed by running `task --list-all`.
+To run unit tests locally:
+```bash
+task test
+```
 
-Set up a local [poetry](https://python-poetry.org/) environment.
+To run a specific test pass pytest arguments to the `test` command:
 
-Make sure to run linter with `task lint-fix`.
+```bash
+task test -- tests/test_alignments.py
+```
 
-For changes in the Taskcluster graph run `task taskgraph-validate` to validate the graph locally.
+Some tests require a compiled Marian and other tools, which is easier to set up with Docker.
 
-## Tests
+It's also possible to run simpler tests using regular `pytest` commands if all the requirements are installed locally.
 
-To run unit tests locally:
+## Docker
+
+Build a Docker image and attach to the running container with:
+
+```bash
+task docker
 ```
+
+Then run commands from inside the container in the same way as from the local environment, for example:
+
+```bash
 task test
 ```
 
+## Updating Python packages
+
+Each pipeline step has its own requirements file with the required packages 
+and a compiled full list of packages generated with `pip-compile` from `pip-tools` (requires `pip install pip-tools`).
+
+For example for the [evaluation step](https://github.com/mozilla/translations/tree/main/pipeline/eval/requirements):
+```
+pipeline/eval/requirements/eval.in
+pipeline/eval/requirements/eval.txt
+```
+
+
+To update the packages, first update the `eval.in` file, then run:
+
+```bash
+task update-requirements -- pipeline/eval/requirements/eval.in
+```
+
+
+## Before committing
+
+Make sure to run linting with `task lint-fix`.
+
+For changes in the Taskcluster graph run `task taskgraph-validate` to validate the graph locally.
+
+
 ## CI
 
 We run all training pipeline steps with a minimal config on pull requests. It runs on the same hardware as a production run.
-Make sure to use `[skip ci]` directive in the PR description not to trigger the run it if not needed to save resources.
+Make sure to use `[skip ci]` directive in the PR description not to trigger the run if it's not intended to save resources.
 If you do run it, minimize pushing to the branch.
 
 Ideally every new push to PR without `[skip ci]` should mean to test the new changes using CI.
 
 We do not run the pipeline on branches without a corresponding pull request.
 
+## Architecture
+
+All the pipeline steps are independent and contain scripts that accept arguments, read input files from disk and output the results to disk.
+It allows writing the steps in any language (currently it's historically mostly bash and Python) and
+represent the pipeline as a directed acyclic graph (DAG).
+
+The DAG of tasks can be launched using any workflow manager
+(currently we support [Snakemake](snakemake.md) and [Taskcluster](task-cluster.md)).
```

Comment:
I'm not sure we should claim to actually support Snakemake. Maybe mention that is unmaintained and we will accept contributions to fix the support.

Reply:
Agreed, it's misleading